### ISSUE TITLE
Stats calculations

### ISSUE THREAD

**Issue by r3dunlop** [cpkm/darts-site#92]
The stats calculations are going weird. It looks like additional copies of players are being created for each season. On top of this, it looks like if you submit stats, edit them, and submit them again, both are added to the PlayerStats. I'm of a mind to destroy the player stats table (or ignore it for now) and run specific queries as we need them. Any reasonable computer should be able to crunch our data fast enough to not notice any latency. If there is latency we can fix the table. Thoughts?

**Comment by r3dunlop** [comment]
- The multiple seasons showing up on the all-time can be fixed by an appropriate aggregation. 
- Note Ryan has 7 GP and Dummy has 16 GP in 2017. There have only been 2 matches entered. My guess is that hitting the submit button does a calculation that doesn't do a rewrite of previous stats for that match.

**Comment by cpkm** [comment]
Ill look at this soon, but Im not surprised. Stats are sepearated by season (so new table entry per player per season) and the update_player_stats needs to be looked at #30 . The issue could just be a display problem, or in the underlying calculation. Need to look

**Comment by r3dunlop** [comment]
How import is this calculation? It makes querying easier in some regards, but also means you have to potentially think a little more when accessing various tables and may be harder to maintain. My personal preference would be query and aggregate from the stats in the games table rather then try to keep reconciliation of the data in two different tables.

**Comment by r3dunlop** [comment]
I'll add some context here. I'm currently working on a data set with 45 k rows. My laptop can display just about any query I throw at it instantaneously. I'm not sure we will ever play 45 k games. The only potential slow down would be a required join, but the player table will be joined on index, will be hashed, and will be very quick as well.

**Comment by cpkm** [comment]
I'm not concerned about performance in this respect. More just that they
are common calculations and you'd be rewriting the same queries many times.
I don't think it's a big difference either way. Though it's probably
beneficial to calculate and keep aggregated stats after each season

**Comment by cpkm** [comment]
Actually im not sure I read your first comment correctly. The stats are
already calculated from the games table, there's no reconciliation or
double entry anywhere.

**Comment by r3dunlop** [comment]
But the data is kept in two places. If I accidentally change the data and don't call "update stats" or whatever it is, won't that cause problems? Rewriting queries isn't a huge deal because you can just make functions for the most used calls. I'm only making this suggestion from a maintenance and debugging point of view.

**Comment by cpkm** [comment]
Ya the calculated/aggregated stats are stored separately. Basically,
anywhere scores or matches are entered/edited should have an update stats
call after. If not they stats would be out of date until it is called.

Do you want to write up a PR? You can probably just pulls most the queries
from the update_player_stats and modify as needed.

**Comment by cpkm** [comment]
I think I know where  the original weirdness was coming from. It's only updating the stats of the roster of the game that scores are submitted for. So if a player (or dummy) was removed from the scores in an edit, their stats wouldn't be updated. I ran an update_player_stats for all players for both seasons and I think it's fixed.

I'm not closing this though because I haven't fixed the underlying logic.

Also note, there currently isn't any logic for handing 'all seasons' on the leaderboard page. I think it just defaults to the most recent.
